kubeyard/commands/shell.py

```diff
@@ -47,7 +51,19 @@ def run_default(self):
     @cached_property
     def pod_name(self) -> str:
         if self.pod:
-            return self.pod
+            all_pods = sh.kubectl.get.pods('-o', 'jsonpath={.items[*].metadata.name}').split()
+            # Exact match
+            if self.pod in all_pods:
+                return self.pod
+            # Starting-with match
+            pods = [pod for pod in all_pods if pod.startswith(self.pod)]
+            if len(pods) == 1:
```

Comment:
We should sort pods before processing them later:
`pods.sort(key=len)`

Reply:
OK